core/src/main/java/org/elasticsearch/rest/action/admin/indices/upgrade/RestUpgradeAction.java

```diff
@@ -86,8 +88,11 @@ public RestResponse buildResponse(UpgradeResponse response, XContentBuilder buil
                 builder.startObject();
                 buildBroadcastShardsHeader(builder, request, response);
                 builder.startObject("upgraded_indices");
-                for (Map.Entry<String, String> entry : response.versions().entrySet()) {
-                    builder.field(entry.getKey(), entry.getValue(), XContentBuilder.FieldCaseConversion.NONE);
+                for (Map.Entry<String, Tuple<Version, String>> entry : response.versions().entrySet()) {
+                    builder.startObject(entry.getKey(), XContentBuilder.FieldCaseConversion.NONE);
+                    builder.field("version", entry.getValue().v1());
+                    builder.field("lucene_version", entry.getValue().v2());
```

Comment:
I am going to rename them to `upgrade_version` and `minimum_compatible_lucene_version`.